Dockerfile

```diff
@@ -0,0 +1,56 @@
+FROM debian:stable as builder
+
+RUN apt-get update && apt-get install -y --no-install-recommends curl git wget ca-certificates
+
+WORKDIR /java/
+
+# Download the JRE for copying to the image to run the OTP server
+RUN curl -Lo jre17.tar.gz https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.7%2B7/OpenJDK17U-jre_x64_linux_hotspot_17.0.7_7.tar.gz
+RUN tar xvf jre17.tar.gz && rm jre17.tar.gz
+
+# Download the JDK and maven and add them to path for building OTP and onebusaway-gtfs-modules
+RUN curl -Lo jdk17.tar.gz https://github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.7%2B7/OpenJDK17U-jdk_x64_linux_hotspot_17.0.7_7.tar.gz
+RUN tar xvf jdk17.tar.gz && rm jdk17.tar.gz
+ENV JAVA_HOME=/java/jdk-17.0.7+7/
+ENV PATH="$JAVA_HOME/bin:$PATH"
+
+RUN curl -Lo maven.tar.gz https://dlcdn.apache.org/maven/maven-3/3.9.1/binaries/apache-maven-3.9.1-bin.tar.gz
+RUN tar xvf maven.tar.gz && rm maven.tar.gz
+ENV PATH="/java/apache-maven-3.9.1/bin/:$PATH"
```

Comment:
Is there a good reason to do that instead of running `apt-get install maven`?

Reply:
Good question, I was just following what the OTP1 Dockerfile was doing, downloading the releases like this did seem a bit strange to me. I'll try using apt and see if I run into any issues.

Reply:
Actually, after looking into it a little bit, I think this is the optimal approach. This looks to be the recommended install method by maven. Using apt-get means relying on the apt repository to have the specific versions we want to use, and there's no great way to ensure that a version is available and can be installed with apt. So downloading the binaries directly is a more straightforward way to make sure we're always using a consistent version. Same goes for the java install, the debian repositories don't reliably have specific versions of java available, just a few recent ones.

Reply:
I am wondering if it is necessary to use a specific version of Maven for our builds. Generally, I would expect that the latest version of Maven would be sufficient to handle our build requirements, as we do not seem to be doing anything out of the ordinary

Reply:
That's fair, it probably isn't necessary to use a specific version, it's not exactly something that has breaking changes that often. But the debian repositories are often quite far behind the latest version, currently it's at [3.6.3](https://packages.debian.org/bullseye/maven), 3.9.1 is the latest.

Reply:
Good point, I haven't realized it would be that outdated.